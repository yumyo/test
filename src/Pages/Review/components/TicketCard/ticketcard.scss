.card {
  padding: 10px;

  &:after {
    border-bottom: 3px solid #f4c4a9;
  }
}

.card.active {
  background: #f4c4a9;
}
