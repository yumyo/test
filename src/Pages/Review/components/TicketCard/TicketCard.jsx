import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addToBasket } from '../../actions';
import './ticketcard.scss';

export class TicketCardComponent extends Component {
    constructor (props) {
        super(props);
        this.state = {};
    }

    handleClick = (item) => {
        this.props.dispatch(addToBasket(item));
        this.setState({
            active: true,
        });
    }

    render () {
        const { product: { id, title, price } } = this.props;
        const activeClass = this.state.active ? 'card active' : 'card';
        return (
            <div
                className={activeClass}
                onClick={() => this.handleClick({id, title, price})}
            >
                <h3>{title}</h3>
                <p>Â£{price}</p>
                <button>Add to Basket</button>
            </div>
        );
    }
}

TicketCardComponent.propTypes = {
    dispatch: PropTypes.func.isRequired,
    product: PropTypes.shape({
        id: PropTypes.string,
        title: PropTypes.string,
        price: PropTypes.decimal,
        quantity: PropTypes.number,
    }).isRequired,
};

export default connect()(TicketCardComponent);
