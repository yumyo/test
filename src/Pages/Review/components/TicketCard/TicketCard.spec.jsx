import React from 'react';
import { shallow } from 'enzyme';
import { TicketCardComponent } from './TicketCard.jsx';

describe('<TicketCard />', () => {
    let component;
    let instance;
    const props = {
        dispatch: jest.fn(),
        product: {
            id: '1',
            title: 'prod 1',
            price: 10.00,
            quantity: 2,
        },
    };

    beforeEach(() => {
        component = shallow(<TicketCardComponent {...props} />);
        instance = component.instance();
    });

    it('renders', () => {
        expect(component).toMatchSnapshot();
    });

    it('calls dispatch with product info', () => {
        component.find('.card').simulate('click');
        expect(props.dispatch).toHaveBeenCalledWith({
            "id": "1",
            "price": 10,
            "title": "prod 1",
            "type": "ADD_TO_BASKET",
        });
    });

    it('updates local state', () => {
        component.find('.card').simulate('click');
        expect(instance.state).toEqual({ active: true});
    });
});
