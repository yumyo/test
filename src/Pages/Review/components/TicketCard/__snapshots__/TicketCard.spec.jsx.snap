// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketCard /> renders 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <h3>
    prod 1
  </h3>
  <p>
    Â£
    10
  </p>
  <button>
    Add to Basket
  </button>
</div>
`;
